### PR TITLE
Number of leaf nodes in binary tree

### PR DESCRIPTION
**Pull request by Purvak-10**
## Binary Tree
- Number of leaf nodes in binary tree, issue number #656 

## I added an algorithms to find:
-Number of leaf nodes in the binary tree
Some outputs with screenshots are given down below.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [ ] 📝 My change requires a change to the README.
- [ ] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [ ] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [ ] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots

Output 1:
  ```
                 30
           /              \
        20                40
     /    \             /      \
  10       25          35      45

```
No. of leaf nodes in tree: 4

Picture1

Output 2:
```
100
  \
   120
      \
      130
         \
         140
            \
            150
              \
              160
                 \
                 170
```
No. of leaf nodes of tree: 1

Picture2

### CODE REVIEW

**Comment by Purvak-10** [comment]
@Lakhankumawat @miraj0507 sir, please check my pull request.

File: T-Tree/B-BinaryTree/No_of_Leaf_Nodes/No_of_leaf_nodes.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#include <bits/stdc++.h>
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/No_of_Leaf_Nodes/No_of_leaf_nodes.cpp:1]: Change File directory to `T-Tree/B-BinaryTree/No_of_leaf_nodes.cpp` and README Changes in the end for your newly added c++ file

File: T-Tree/B-BinaryTree/No_of_Leaf_Nodes/No_of_leaf_nodes.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+{
+
+    struct Node *p = new Node;
+    p->value = val;
+    p->left = p->right = NULL;
+    p->root = NULL;
+    return p;
+}
+
+struct Node* tree(Node* t, int v){
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/No_of_Leaf_Nodes/No_of_leaf_nodes.cpp:23]: Add comments in code

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+<!-- follow the template of Bubble Sort, add the respective heading in Table of content -->
+
+
+<!-- Table of content -->
+# Table of content
+- Number of leaf nodes in Binary Tree.
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md:6]: Your Table of content should be like this https://github.com/Lakhankumawat/LearnCPP/blob/main/T-Tree/G-GenricTree/README.md

File: T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                 30
+           /              \
+        20                40
+     /    \             /      \
+  10       25          35      45
+```
+No. of leaf nodes in tree: 4
+
+![Picture1](https://user-images.githubusercontent.com/76229635/159126054-e544d3af-2302-4b33-aa07-1645973dd3f6.png)
+
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md:69]: Remove these preview and output screenshots instead you need to show only one working image of your code 
See here https://github.com/Lakhankumawat/LearnCPP/blob/main/T-Tree/G-GenricTree/README.md

File: T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+### Properties
+
+- Time Complexity : O(n)
+
```
Comment (Lakhankumawat) [inline T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md:74]: You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md (hunk 4 above)
Comment (Purvak-10) [reply T-Tree/B-BinaryTree/No_of_Leaf_Nodes/Readme.md:69]: @Lakhankumawat sir, how to show working of process, should i explain it?

**Comment by Purvak-10** [comment]
@miraj0507 sir, please check the pr i have made the changes

File: T-Tree/B-BinaryTree/Readme.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+```
+No. of leaf nodes in tree: 4
+
+![Picture1](https://user-images.githubusercontent.com/76229635/159126054-e544d3af-2302-4b33-aa07-1645973dd3f6.png)
+
+
+### Properties
+
+- Time Complexity: O(n)
+
```
Comment (siddhi-244) [inline T-Tree/B-BinaryTree/Readme.md:137]: Add space Complexity too

Comment (Purvak-10) [reply T-Tree/B-BinaryTree/Readme.md:139]: @siddhi-244 please review this, i have done the editing.

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
Try to link the table of contents to actual headings. You can refer other prs !

**Comment by Lakhankumawat** [comment]
You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by Lakhankumawat** [comment]
Instead of output screenshot in README add working images

**Comment by Purvak-10** [comment]
> Instead of output screenshot in README add working images

what is mean of working images? can you please elaborate it

**Comment by Purvak-10** [comment]
> Try to link the table of contents to actual headings. You can refer other prs !

@siddhi-244 please review, i have done the editing & mam "what is working images?". If it mean about inputs so for that instead of photos, i have already put in it in words because i think that won't look good if i illustrate my example on a paper.

**Review by siddhi-244 — APPROVED** [review]
LGTM

**Review by Lakhankumawat — APPROVED** [review]
LGTM 🎉✨

**Comment by Lakhankumawat** [comment]
@all-contributors please add @Purvak-10 code ,

**Pull request merged**